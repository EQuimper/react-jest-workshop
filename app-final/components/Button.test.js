import React, {PropTypes, Component} from 'react'
import renderer from 'react-test-renderer'
import Button from './Button'

test('styles the button with a background of the context color', () => {
  const component = renderer.create(
    getContextProvider(<Button>Click Me</Button>, {color: 'blue'})
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

function getContextProvider(children, context) {
  class ContextProvider extends Component {
    getChildContext() {
      return context
    }
    render() {
      return <div>{children}</div>
    }
  }
  ContextProvider.childContextTypes = Object.keys(context).reduce((obj, key) => {
    obj[key] = PropTypes.any
    return obj
  }, {})
  return <ContextProvider />
}
