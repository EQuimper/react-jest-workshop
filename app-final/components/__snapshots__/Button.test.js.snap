exports[`test styles the button with a background of the context color 1`] = `
<div>
  <button
    style={
      Object {
        "background": "blue"
      }
    }>
    Click Me
  </button>
</div>
`;
